attend/attend.php

```diff
@@ -0,0 +1,388 @@
+<?php
+  // todo; this should really be based on chapter
+  // Although it looks like airtable doesn't care about timezones?
+  date_default_timezone_set('America/Los_Angeles');
+
+  session_start();
+  require_once ('google-api-php-client/src/Google/autoload.php');
+
+  require_once('attend/googleCredentials.php');
+  $redirect_uri  = 'http://' . $_SERVER['HTTP_HOST'] . '/attend/';
+
+  $client = new Google_Client();
+  $client->setClientId($client_id);
+  $client->setClientSecret($client_secret);
+  $client->setRedirectUri($redirect_uri);
+  $client->setScopes('email');
+  $client->addScope(Google_Service_Directory::ADMIN_DIRECTORY_GROUP_MEMBER_READONLY);
+
+  /************************************************
+    Service user authentication
+    Service user is necessary for figuring out group membership
+   ************************************************/
+  try {
+    if (strpos($service_client_id, "googleusercontent") == false
+        || !strlen($service_account_name)
+        || !strlen($service_key_file_location)) {
+      echo missingServiceAccountDetailsWarning();
+      exit;
+    }
+    
+    $service_client = new Google_Client();
+    $service_client->setApplicationName("DxE Attendance");
+    $book_service = new Google_Service_Books($service_client);
+    $directory_service = new Google_Service_Directory($service_client);
+    
+    /************************************************
+      If we have an access token, we can carry on.
+      Otherwise, we'll get one with the help of an
+      assertion credential. In other examples the list
+      of scopes was managed by the Client, but here
+      we have to list them manually. We also supply
+      the service account
+     ************************************************/
+    if (isset($_SESSION['service_token'])) {
+      $service_client->setAccessToken($_SESSION['service_token']);
+    }
+    $service_key = file_get_contents($service_key_file_location);
+    $service_cred = new Google_Auth_AssertionCredentials(
+        $service_account_name,
+        array(Google_Service_Directory::ADMIN_DIRECTORY_GROUP_MEMBER_READONLY),
+        $service_key
+    );
+    // impersonate a super admin
```

Comment:
It's a google apps super admin (https://support.google.com/a/answer/33325?hl=en)